File: application/src/main/java/whenyourcar/application/facade/chat/ChatMessageFacade.java
package whenyourcar.application.facade.chat;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import whenyourcar.application.dto.chat.message.ChatMessageResponse;
import whenyourcar.application.service.chat.ChatMessageService;
import whenyourcar.application.service.chat.ChatRoomService;
import whenyourcar.application.service.user.UserCommonService;
import whenyourcar.common.code.exception.GeneralException;
import whenyourcar.common.code.status.ErrorStatus;
import whenyourcar.domain.entity.User;

@Service
@RequiredArgsConstructor
public class ChatMessageFacade {
    private final UserCommonService userCommonService;
    private final ChatRoomService chatRoomService;
    private final ChatMessageService chatMessageService;
    public ChatMessageResponse.ChatMessageResponseSetDto getChatMessage(String email,
                                                                        Long roomId,
                                                                        Pageable pageable) {
        User user = userCommonService.getUserByEmail(email);
        if (chatRoomService.isUserInRoom(roomId, user.getId())) {
            return chatMessageService.getChat(pageable, roomId, user.getId());
        }else {
            throw new GeneralException(ErrorStatus.ROOM_IS_NOT_EXIST);
        }
    }

}

File: application/src/main/kotlin/howmuchcar/application/facade/chat/ChatMessageFacade.kt
package howmuchcar.application.facade.chat

import howmuchcar.application.dto.chat.ChatMessageResponse
import howmuchcar.application.port.`in`.chat.ChatMessageUseCase
import howmuchcar.application.port.`in`.chat.ChatRoomUseCase
import howmuchcar.application.port.`in`.user.UserCommonUseCase
import howmuchcar.common.exception.RestApiException
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class ChatMessageFacade (
    private val chatRoomUseCase: ChatRoomUseCase,
    private val chatMessageUseCase: ChatMessageUseCase,
    private val userCommonUseCase: UserCommonUseCase
){
    fun getChatMessage(
        user: User,
        roomId: Long,
        pageable: Pageable
    ): ChatMessageResponse {
        chatRoomUseCase.isUserInRoom(roomId, user.id)
        return chatMessageUseCase.getChat(pageable, roomId, user.id)
    }

    fun sendMessageToQueue(
        roomId: Long,
        senderId: Long,
        message: String
    ) {
        val sender = userCommonUseCase.getUserById(senderId)
        val chatId = chatMessageUseCase.saveChat(roomId, sender, message)
        chatMessageUseCase.sendMessageToMessageQueue(roomId, chatId, sender.id, message)
    }
}